mage_ai/data_preparation/models/pipeline.py

```diff
@@ -1106,6 +1167,87 @@ async def update(self, data, update_content=False):
                     cache.remove_pipeline(tag_uuid, old_uuid)
                 cache.add_pipeline(tag_uuid, self)
 
+    # Updates pipeline YAML for new pipeline or block names from the import process
+    # compiles list of files to be written
+    # also returns the new pipeline name and the config destination path
+    def __update_pipeline_yaml(self, tmp_dir, config_zip_path, overwrite):
+        files_to_be_written = []
+        with open(config_zip_path, 'r') as pipeline_config:
+            config = yaml.safe_load(pipeline_config)
+
+            # check if pipeline exists with same uuid and generate new one if necessary
+            if not overwrite:
+                uuid = config['uuid']
+                index = 0
+                while self.exists(uuid):
+                    index += 1
+                    uuid = f'{config["uuid"]}_{index}'
```

Comment:
I'm worried that using actual uuids would not result in a good user experience since these uuids would show up on the UI. Nevertheless we could definitely make this change, I just want to make sure we first view this issue from all angles.